client/gutenberg/extensions/mailchimp/edit.jsx

```diff
@@ -182,36 +184,43 @@ class MailchimpSubscribeEdit extends Component {
 				</PanelBody>
 			</InspectorControls>
 		);
+		const emailInputId = `${ classPrefix }email_input-${ instanceId }`;
```

Comment:
Is this only for `label`'s `for` (and the corresponding `id`/`name`)? If so, there's an [alternative `label` style](https://developer.mozilla.org/de/docs/Web/HTML/Element/label) you could consider:

> There is another style to consider. You can nest the <input> directly inside the <label>, in which case the for/id attributes are not needed because the association is implicit:
>
> ```jsx
> <label>Do you like peas?
>  <input type="checkbox" name="peas">
> </label>
> ```

Reply:
Since we're hiding the label I was under the impression this wasn't a viable pattern — is this not so?

> s this only for `label`'s `for` (and the corresponding `id`/`name`)? If so, there's an [alternative `label` style](https://developer.mozilla.org/de/docs/Web/HTML/Element/label) you could consider:

Since we're hiding the label I was under the impression this wasn't a viable pattern — is this not so?

Reply:
Yet another alternative, using `aria-label` attribute in input: https://github.com/Automattic/wp-calypso/pull/30964/files

Reply:
> Since we're hiding the label I was under the impression this wasn't a viable pattern — is this not so?

Uh right, I missed that. I'm not sure TBH. I mean semantically it's still a label -- it's just that we're hiding it visually. (My line of argument is a bit hand-waving here -- are there any established rules for this kind of question?)

Using only `aria-label` seems to make sense to me, but I'm not sure that's permissible, a11y-wise.

Reply:
We do have `aria-label` in the input on the frontend, which is where it really matters. In the editor, the input is disabled anyway:

https://github.com/Automattic/jetpack/pull/11403/files#diff-e800fe1ba24cea2c2da6f6ac9f542d4dR79

Reply:
`aria-label` is one of the suggested ways to hide label in https://www.w3.org/WAI/tutorials/forms/labels/#using-aria-label

Reply:
As per https://github.com/Automattic/wp-calypso/pull/30970#discussion_r260152768 I removed redundant label element in https://github.com/Automattic/wp-calypso/pull/30970/commits/2477246136fb1bc97ba0b68102f87c162da0a5a8 and left `aria-label` attribute.